python/tests/kat/t_tcpmapping.py

```diff
@@ -264,3 +265,735 @@ def check(self):
 
             assert backend_fqdn == wanted_fqdn, f'{idx}: backend {backend_fqdn} != expected {wanted_fqdn}'
             assert tls_enabled == tls_wanted, f'{idx}: TLS status {tls_enabled} != wanted {tls_wanted}'
+
+
+class TCPMappingBasicTest(AmbassadorTest):
+    extra_ports = [ 6789 ]
+    target: ServiceType
+
+    def init(self) -> None:
+        self.target = HTTP()
+
+    def manifests(self) -> str:
+        return format('''
+---
+apiVersion: getambassador.io/v2
+kind: TCPMapping
+metadata:
+  name: {self.name.k8s}
+spec:
+  ambassador_id: [ {self.ambassador_id} ]
+  port: 6789
+  service: {self.target.path.fqdn}:80
+''') + super().manifests()
+
+    def queries(self):
+        yield Query(self.url("", port=6789))
+
+    def check(self):
+        assert self.results[0].json["backend"] == self.target.path.k8s
+        assert self.results[0].json["request"]["tls"]["enabled"] == False
+
+
+class TCPMappingCrossNamespaceTest(AmbassadorTest):
+    extra_ports = [ 6789 ]
+    target: ServiceType
+
+    def init(self) -> None:
+        self.target = HTTP(namespace="other-namespace")
+
+    def manifests(self) -> str:
+        return namespace_manifest("other-namespace") + format('''
+---
+apiVersion: getambassador.io/v2
+kind: TCPMapping
+metadata:
+  name: {self.name.k8s}
+spec:
+  ambassador_id: [ {self.ambassador_id} ]
+  port: 6789
+  service: {self.target.path.fqdn}:80
+''') + super().manifests()
+
+    def queries(self):
+        yield Query(self.url("", port=6789))
```

Comment:
For my understanding: 

Does `Query` just send a random ping down a tcp socket? Then the results happens to be an http response which is interpreted and plugged into the results? Or is it sending an http request (since its built on tcp) which is enough to say that envoy is listening for TCP traffic and proxying it to the backend without considerations for the L7 protocol?

Reply:
It's sending an HTTP request.

This test works because the `TCPMapping` points to something that is itself serving HTTP.  Another one of the tests pints at something that is itself serving `HTTPS`, and we check the cert that it uses in order to test that Emissary is passing the raw stream through rather than terminating and re-originating the TLS.

Reply:
Got it makes sense. Thanks!